edulution/offline_testing
# bug fix for index of topics

### Summary


This PR fixes Issue #13 
- The data for the updated index of topics page is now dynamically assigned by a database view that queries for all courses in the baseline _testing database therefore fixing the scalability concern.
- The JSON file is populated by an api endpoint by fetching content from the database view created above.
- Previous Angular Module for index of topics has been updated to use the endpoint above above.
- The data from the module is bound to the front end.
- Live search and pagination  have been update to correctly reflect the displayed topics when a search is performed.

**The page displays the topics and their location. Additionally, the pagination correctly reflects the number of pages with content .**

![sc0](https://user-images.githubusercontent.com/59157194/125812632-159a0a59-5048-481a-8967-6847031676ce.png)
![sc01](https://user-images.githubusercontent.com/59157194/125812501-46406833-b5c4-47a3-be27-075aeaf29349.png)

**Live search displays matching content and pagination reflects the amount of content displayed. If the content is more than the maximum display value per page, the page will show more pages.**
![sc02](https://user-images.githubusercontent.com/59157194/125812505-aa98ace4-e6ad-4ab7-80d6-7c27078a7664.png)


### Consequences for the user.
"The Index of Topics page will always be accurate and be a direct reflection of the topics in the Kolibri courses on the device."
…

### Reviewer guidance



- Pull the latest code and run flyway migrations
- Open Index of Topics page
- Perform a search and observe whether the content  is updated as one types.
- Observe if the pagination reflects the number of contents displayed. 

…
